### PR TITLE
V2 test

### PR DESCRIPTION
**Pull request by vivianjeng**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): test scripts

## What is the current behavior?

Issue Number:

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/contracts/test/attestation.test.ts [hunk 1]
```diff
@@ -35,101 +31,86 @@ describe('Attestations', function () {
...
     })
 
     it('should fail to submit from non-attester', async () => {
         const accounts = await ethers.getSigners()
-        const id = new ZkIdentity()
-        const epochRoot = await unirepContract.attesterEpochRoot(
-            accounts[1].address,
-            0
+        const attester = accounts[1]
+        const wrontAttester = accounts[5]
```
Comment (vimwitch) [inline packages/contracts/test/attestation.test.ts:88]: typo here

File: packages/contracts/test/attestation.test.ts [hunk 2]
```diff
@@ -138,33 +119,114 @@ describe('Attestations', function () {
...
                 epochKey,
-                accounts[1].address,
+                attester.address,
                 posRep,
                 negRep,
                 graffiti,
                 timestamp
             )
+        {
+            const tx = await unirepContract.buildHashchain(
```
Comment (vimwitch) [inline packages/contracts/test/attestation.test.ts:131]: Why are we doing the hashchain logic here? We've already tested what we're looking for

File: packages/core/test/userState.test.ts [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        await userState.waitForSync()
+        const newEpoch = await userState.latestTransitionedEpoch()
+        expect(newEpoch).equal(oldEpoch + 1)
+        await userState.stop()
+    })
+
+    // TODO: should fix reputation proof first
+    // it('reputation proof', async () => {
+    //     const accounts = await ethers.getSigners()
+    //     const attester = accounts[1]
```
Comment (vimwitch) [inline packages/core/test/userState.test.ts:191]: We have a functioning reputation proof. I think we should separate the spending proof and the rep proof though.

File: packages/contracts/test/attestation.test.ts (hunk 2 above)
Comment (vivianjeng) [reply packages/contracts/test/attestation.test.ts:131]: `buildHashchain` is not checked in `@unirep/contracts`
I think `@unirep/contracts` can cover all the functions in `Unirep.sol`

Comment (vimwitch) [reply packages/contracts/test/attestation.test.ts:131]: Ok but I think that should be in a different test case

Comment (vivianjeng) [reply packages/contracts/test/attestation.test.ts:131]: Create the issue here https://github.com/Unirep/Unirep/issues/163
We can do that in another PR

Comment (vimwitch) [reply packages/contracts/test/attestation.test.ts:131]: Ok, in the meantime we should take out this extra test logic.

**Review by vimwitch — APPROVED** [review]

**Pull request merged**